File: ParkPal.API/ParkPal.API/Models/Requests/Notification/CreateNotificationRequest.cs
using ParkPal.Common.Models.Database.Entities.Notification.Enums;
using Type = ParkPal.Common.Models.Database.Entities.Notification.Enums.Type;

namespace ParkPal.API.Models.Requests.Notification;

public class CreateNotificationRequest
{
    public string? AttractionId { get; set; }
    public string ParkId { get; set; }
    public Type Type { get; set; }
    public int MinuteInterval { get; set; }
    public CriteriaType CriteriaType { get; set; }
    public int WaitTime { get; set; }
}

File: ParkPal.Application/src/models/api/requests/notification/CreateNotificationRequest.ts
export default class CreateNotificationRequest {
    attractionId?: string | null;
    type?: number;
    parkId?: string;
    minuteInterval?: number = 5;
    criteriaType?: number;
    waitTime?: number;

    constructor(data: Pick<CreateNotificationRequest, "attractionId" | "type" | "parkId" | "minuteInterval" | "criteriaType" | "waitTime"> | null = null) {
        if(data != null) {
            this.attractionId = data.attractionId;
            this.type = data.type;
            this.parkId = data.parkId;
            this.minuteInterval = data.minuteInterval;
            this.criteriaType = data.criteriaType;
            this.waitTime = data.waitTime;
        }
    }
}